src/core/structure.js

```diff
@@ -308,6 +312,10 @@ p5.prototype.pop = function() {
  *
  */
 p5.prototype.redraw = function(n) {
+  if (this._inUserDraw || !this._setupDone) {
```

Comment:
Note that this breaks calling `redraw` from within `draw` manually, which could be useful. However, I'd be willing to say that that is not an important use case.

Reply:
indeed. although the docs above this do state that you probably shouldn't be calling it in `draw` anyway.